### PR TITLE
HOMME: Make C++ limiter_option=9 impl BFB with Fortran with cycle sta…

### PR DESCRIPTION
**Pull request by ambrad**
…tement.

### CODE REVIEW

File: components/homme/src/share/cxx/EulerStepFunctorImpl.hpp [hunk 1]
```diff
@@ -994,29 +1000,66 @@ VECTOR_SIMD_LOOP
...
       }
     }
+
+    forij {
+      VECTOR_SIMD_LOOP forlev {
+        x(i,j,lev) *= dpmass(i,j,lev);
+      }
+    }
   } else if (limiter_option == 9) {
+    ViewUnmanaged<Real[NP][NP][NUM_PHYSICAL_LEV]>
```
Comment (ambrad) [inline components/homme/src/share/cxx/EulerStepFunctorImpl.hpp]: This should be NUM_LEV*VECTOR_SIZE.

File: components/homme/src/share/cxx/EulerStepFunctorImpl.hpp [hunk 2]
```diff
@@ -994,29 +1000,66 @@ VECTOR_SIMD_LOOP
...
+      ptens(&iptens(0,0,0)[0]);
+    ViewUnmanaged<Real[NP][NP][NUM_PHYSICAL_LEV]>
```
Comment (ambrad) [inline components/homme/src/share/cxx/EulerStepFunctorImpl.hpp]: This doesn't have to be, but do so for alignment.

File: components/homme/src/share/cxx/EulerStepFunctorImpl.hpp [hunk 3]
```diff
@@ -879,54 +884,59 @@ VECTOR_SIMD_LOOP
...
-# define forij for (int i = 0; i < NP; ++i) for (int j = 0; j < NP; ++j)
-# define forlev for (int lev = 0; lev < NUM_PHYSICAL_LEV; ++lev)
+#define forij for (int i = 0; i < NP; ++i) for (int j = 0; j < NP; ++j)
+#define forlev for (int lev = 0; lev < NUM_PHYSICAL_LEV; ++lev)
 
   ViewUnmanaged<const Real[NP][NP][NUM_LEV*VECTOR_SIZE]>
     dpmass(&idpmass(0,0,0)[0]);
-  ViewUnmanaged<Real[NP][NP][NUM_LEV*VECTOR_SIZE]>
-    x(&iptens(0,0,0)[0]), c(&irwrk(0,0,0)[0]);
+  ViewUnmanaged<Real[NP][NP][NUM_PHYSICAL_LEV]>
```
Comment (ambrad) [inline components/homme/src/share/cxx/EulerStepFunctorImpl.hpp]: Change this one, too, for alignment.

File: components/homme/test_execs/share_kokkos_ut/CMakeLists.txt [hunk 4]
```diff
@@ -163,6 +163,47 @@ ELSE()
...
+)
+SET (LIMITERS_UT_CXX_SRCS
+  ${SRC_SHARE_DIR}/cxx/EulerStepFunctorImpl.hpp
+  ${SRC_SHARE_DIR}/cxx/Hommexx_Session.cpp
+  ${SRC_SHARE_DIR}/cxx/mpi/Comm.cpp
+  ${SRC_SHARE_DIR}/cxx/ExecSpaceDefs.cpp
+  ${SHARE_UT_DIR}/limiters.cpp
+)
+
+SET (CONFIG_DEFINES NP=4 NC=4 PLEV=16 QSIZE_D=4 _MPI=1 _PRIM HOMMEXX_NO_CONTEXT HOMMEXX_NO_MPI_CONTEXT)
```
Comment (ambrad) [inline components/homme/test_execs/share_kokkos_ut/CMakeLists.txt]: For better testing of pack arithmetic, change PLEV to, say, 13.

File: components/homme/src/share/cxx/EulerStepFunctorImpl.hpp (hunk 2 above)
Comment (bartgol) [reply components/homme/src/share/cxx/EulerStepFunctorImpl.hpp]: Isn't it already aligned?

File: components/homme/test_execs/share_kokkos_ut/CMakeLists.txt (hunk 4 above)
Comment (bartgol) [reply components/homme/test_execs/share_kokkos_ut/CMakeLists.txt]: Good point

**Comment by ambrad** [comment]
I'll have these changes in later today.

**Comment by ambrad** [comment]
OK, fixed. You might mod PLEV for the other unit tests in the same way to test pack arithmetic.

**Comment by bartgol** [comment]
The two commits solving the limiter 9 issues are going to be incorporated in PR #2531 directly. Closing this.

**Pull request closed without merging**